src/Events.ts

```diff
@@ -38,12 +39,12 @@ export function combineEvents<
     return Object.assign({}, ...args)
 }
 
-export function createEventBus<C extends EventDeclaration>(args: { events: C, channels?: Channel[] }): C {
+export function createEventBus<C extends EventDeclaration>(args: { events: C, channels?: Channel[] }): EventDeclaration {
```

Comment:
Wasn't sure about this, ~recent TS versions are complaining

Reply:
Might just need to give `{}` a type before using it in the reduce.

```ts
function createEventBus<C ...>(args: { events: C ... }): C {
  const eventBus: C = {};
  return ...reduce((acc: C, slotName) => { ... }, eventBus);
 }
```

(`...` are just me being lazy, not spread operators)

Reply:
I gave it a try initially, couldn't get it to stop complaining though didn't spend too much time on it.

In the mean time, open to other suggestions, or confirmation that what I did here is fine.

Reply:
it's okay